src/Java.Interop.Tools.JavaCallableWrappers/Java.Interop.Tools.JavaCallableWrappers/CallableWrapperWriterOptions.cs

```diff
@@ -0,0 +1,10 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Java.Interop.Tools.JavaCallableWrappers;
+
+public class CallableWrapperWriterOptions
```

Comment:
What I don't immediately understand is the criteria used to determine which "options" go into `CallableWrapperWriterOptions` and which "options" become properties on the given type.

Consider https://github.com/xamarin/xamarin-android/pull/8701/files, which provides "context" for how these changes would be used within xamarin-android.

*Every property set on `jcw_type`* is the same for *every instance* of `jcw_type`.

Wouldn't it make more sense for the "invariant" properties to instead be on `CallableWrapperWriterOptions`, and then "hoist" the `var options = new CallableWrapperWriterOptions` *before* the `foreach` loop?  This would make explicit the values which don't depend upon type being emitted, as they would be the same for *all* types, e.g.

```csharp
var options = new CallableWrapperWriterOptions {
    ApplicationJavaClass = ApplicationJavaClass,
    CodeGenerationTarget = JavaPeerStyle.XAJavaInterop1,
    GenerateOnCreateOverrides = generateOnCreateOverrides,
    MonoRuntimeInitialization = monoInit,
};
foreach (JavaType jt in newJavaTypes) {
    …
    var jcw_type = CecilImporter.CreateType (t, cache, methodClassifier: classifier);
    jcw_type.Generate (writer, options);
}
```

As a quick spot-check, `CallableWrapperType.cs` use `ApplicationJavaClass`, `GenerateOnCreateOverrides`, and `MonoRuntimeInitialization` from methods which already have a `CallableWrapperWriterOptions options` parameter, so such a change wouldn't be significant.

So why not have these "invariant" properties on `CallableWrapperWriterOptions`?

Reply:
Counter-argument: `CallableWrapperWriterOptions` contains options that apply to Types, Fields, Methods, etc., and e.g. `ApplicationJavaClass` isn't used by `CallableWrapperField`.

Counter-counter-argument: `CecilImporter.CreateType()` is the entrypoint.  If we ever needed to introduce a global MSBuild option specific to field generation, having a caller iterate over all fields within a `CallableWrapperField` to set a field-specific value would be incredibly annoying.  It would be much easier to add such options to `CallableWrapperWriterOptions` directly.  (Or pass it into `CecilImporter.CreateType()`?  At which point we might instead want a `CallableWrapperImportOptions`?)

Reply:
Created `CallableWrapperReaderOptions`, as it seemed like the cleanest way to implement this.